pkgs/development/libraries/wolfssl/default.nix

```diff
@@ -5,37 +5,63 @@
 , autoreconfHook
 , util-linux
 , openssl
+# The primary --enable-XXX variant. 'all' enables most features, but causes build-errors for some software,
+# requiring to build a special variant for that software. Example: 'haproxy'
+, variant ? "all"
+, extraConfigureFlags ? []
+, enableLto ? true
```

Comment:
Looks like you'll have to disable this for darwin x86_64. configure fails even the simple compiler check with it.

Reply:
Failed on darwin aarch64 too, the linker doesn't seem to like LTO. Clang supports LTO as well as GCC, but because I haven't tested it, I disabled LTO for Clang as well.